tapdb/interfaces.go

```diff
@@ -100,6 +101,7 @@ func NewTransactionExecutor[Querier any](db BatchedQuerier,
 func (t *TransactionExecutor[Q]) ExecTx(ctx context.Context,
 	txOptions TxOptions, txBody func(Q) error) error {
 
+txStart:
```

Comment:
This will cause a new `defer` with a rollback to be caused for each retry. And IIUC then `Rollback()` will return an error if it is called twice. So I think we need to roll this out into an actual loop, otherwise we might get weird side effects or errors.